### ISSUE TITLE
Databases and clusters should be created in the same namespace as the TPR

### ISSUE THREAD

**Issue by pst** [CrunchyData/postgres-operator#8]
I have two clusters defined as TPRs. One in the `default` namespace and one in the `pgtest` namespace.

```
$ kubectl get pgcluster --all-namespaces
NAMESPACE   NAME               KIND
default     pg-orange-jaguar   PgCluster.v1.crunchydata.com
pgtest      pg-pink-whale      PgCluster.v1.crunchydata.com

```
However, all deployments are created in the `default` namespace.

```
$ kubectl get deploy --all-namespaces
NAMESPACE         NAME                          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
default           pg-orange-jaguar              1         1         1            0           35m
default           pg-orange-jaguar-replica      2         2         2            0           35m
default           pg-pink-whale                 1         1         1            0           15m
default           pg-pink-whale-replica         2         2         2            0           15m
```
It's not clear to me what the model chosen here is. Do I need one operator per pgcluster, one operator per k8s namespaces or one operator per k8s cluster? Optimally I think the operator should only need to be deployed once per k8s cluster and then I can create multiple pgclusters in multiple namespaces.

**Comment by jmccormick2001** [comment]
thanks, right now the operator is designed to be run in a single namespace and service that namespace, but I agree with you that having it be configured to run against all namespaces would be a nice feature to have, some will want this and others will not I assume.  I think its a valid feature to support.

**Comment by pst** [comment]
Thanks for your quick reply. My feedback is two-fold though. The long term aspect but also the currently it seems broken. Deploying the operator in some other namespace than `default` still puts the cluster in `default`. I think it's because the Deployment metadata has no namespace in the template (https://github.com/CrunchyData/postgres-operator/blob/master/conf/postgres-operator/cluster/1/cluster-deployment.json#L4). Unless I'm missing something.

**Comment by jmccormick2001** [comment]
ah, thanks, appears to be a bug, it should adjust to the namespace defined in the config

**Comment by StephenSorriaux** [comment]
Namespace used by postgres-operator to create pg deployment and replicas is the env var "NAMESPACE" of the postgres-operator deployment which refers to its metadata.namespace: https://github.com/CrunchyData/postgres-operator/blob/master/examples/operator/deployment.json#L24
It should take the namespace where the postgres-operator deployment is created, that is why you need 1 postgres-operator deployment per namespace.

I'm not sure why it is not the case here, I will take a look at this problem

**Comment by jmccormick2001** [comment]
I placed some changes into master that should fix this, everything should run in a single namespace...FYI, I'm planning a new release on Friday if all goes as planned....

**Comment by jmccormick2001** [comment]
should be fixed in the latest release (1.1).
